file/file-alfresco/src/main/java/uk/gov/justice/services/file/alfresco/requester/AlfrescoFileRequester.java

```diff
@@ -0,0 +1,60 @@
+package uk.gov.justice.services.file.alfresco.requester;
+
+import static javax.ws.rs.core.MediaType.valueOf;
+import static javax.ws.rs.core.Response.Status.OK;
+import static uk.gov.justice.services.common.http.HeaderConstants.ID;
+
+import uk.gov.justice.services.common.configuration.GlobalValue;
+import uk.gov.justice.services.file.alfresco.rest.AlfrescoRestClient;
+import uk.gov.justice.services.file.api.FileData;
+import uk.gov.justice.services.file.api.requester.FileRequester;
+
+import java.util.Collections;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.ws.rs.core.MultivaluedHashMap;
+import javax.ws.rs.core.Response;
+
+@ApplicationScoped
+public class AlfrescoFileRequester implements FileRequester {
+    private static final String URL_SEPARATOR = "/";
+
+    @Inject
+    @GlobalValue(key = "alfrescoWorkspacePath", defaultValue = "/service/api/node/content/workspace/SpacesStore/")
+    String alfrescoWorkspacePath;
+
+    @Inject
+    @GlobalValue(key = "alfrescoUploadUser")
+    String alfrescoUploadUser;
+
+    @Inject
+    AlfrescoRestClient restClient;
+
+    @Override
+    public byte[] request(final FileData fileData, final String fileName) {
+        final MultivaluedHashMap<String, Object> headers = new MultivaluedHashMap<>();
+        headers.put(ID, Collections.singletonList(alfrescoUploadUser));
+
+        Response response = restClient.get(buildAlfrescoUri(fileData, fileName), valueOf(fileData.getFileMimeType()), headers);
+
+        final byte[] document = response.readEntity(byte[].class);
+
+        if (response.getStatus() == OK.getStatusCode()) {
+            return document;
+        } else {
+            return null;
+        }
+    }
+
+    private String buildAlfrescoUri(final FileData fileData, final String fileName) {
+        final StringBuilder requestBuilder = new StringBuilder();
```

Comment:
Does it make sense to use the FileData type if you just want to pass fileId to the method?